### ISSUE TITLE
Test suite breaks with latest phpunit 6.4.2

### ISSUE THREAD

**Issue by belgattitude** [doctrine/dbal#2892]
FYI, Travis build breaks after upgrade to phpunit 6.4.2. This test will fail:

```
1) Doctrine\Tests\DBAL\ConnectionTest::testRethrowsOriginalExceptionOnDeterminingPlatformWhenConnectingToNonExistentDatabase
Exception: Original exception
```

**Comment by PowerKiKi** [comment]
In f5738295c592f4228045ba987725d8347d0ed7a6 @JeroenDeDauw ignored composer.lock from git. To me it looks like a bad practice and this issue would have been avoided if the lock file was still in git. I couldn't find any rationale behind ignoring that file. Is there any ?

If not, I'd suggest to re-add `composer.lock` to guarantee a stable test environnement and avoid breaking all PRs at once for unrelated changes.

**Comment by JeroenDeDauw** [comment]
Thanks for pinging.

If you pin the production code dependencies then the CI won't break if the latest version of packages (within the version ranges specified in `composer.json`) stops being compatible. The users of your library will not be so fortunate, and likely prefer if your CI breaks before theirs does.

**Comment by Ocramius** [comment]
Please don't commit the `composer.lock` in library code: we want to test
against the dependency set that our users will actually get, not some
random past state that "worked for us".

We already test max/min deps and that is sufficient to expose these issues.

**Comment by belgattitude** [comment]
Fixed in latest 6.4.3 ;)
